public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -172,18 +214,17 @@ export class LokiDatasource extends DataSourceApi<LokiQuery, LokiOptions> {
       .map(target => this.prepareLiveTarget(target, options));
 
     for (const liveTarget of liveTargets) {
-      const subscription = webSocket(liveTarget.url)
+      liveTarget.subscription = webSocket(liveTarget.url)
```

Comment:
It was kinda unclear to me that we actually keep the buffer state in the `liveTargets` right now. Probably needs a better name (`liveStreams`/`streamState` maybe?) or a comment at least.